src/py-opentimelineio/opentimelineio/url_utils.py

```diff
@@ -40,4 +40,19 @@ def filepath_from_url(urlstr):
     """ Take a url and return a filepath """
 
     parsed_result = urlparse.urlparse(urlstr)
-    return request.url2pathname(parsed_result.path)
+
+    # Check if original urlstr is URL encoded
+    if urlparse.unquote(urlstr) != urlstr:
+        filepath = request.url2pathname(parsed_result.path)
+
+    # Otherwise, combine the netloc and path
+    else:
+        filepath = parsed_result.netloc + parsed_result.path
+
+    filepath = filepath.replace("\\", "/")
+
+    # If on Windows, remove the first leading slash left by urlparse
+    if os.name == 'nt' and filepath.startswith('/'):
+        filepath = filepath[1:]
```

Comment:
@apetrynet, it occurs to me that since this is an OS dependent check it will cause the unit tests I made to fail. Any suggestions on how to test Windows paths properly? I could make a regex that looks for a something like `/S:/` before omitting the leading slash but I'm not sure if a colon is ever used in a Linux or Mac path since I do not typically work in those operating systems.

Reply:
Nvm, I looked into it further and colons are not used in Unix paths so a regex seemed like the best approach for testing locally can and in CI/CD